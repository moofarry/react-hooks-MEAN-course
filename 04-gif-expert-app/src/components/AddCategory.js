import React from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = React.useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prevCategories) => [ inputValue,...prevCategories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        type="text"
        onChange={handleInputChange}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
