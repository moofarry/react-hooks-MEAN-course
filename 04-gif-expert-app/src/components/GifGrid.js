import React from "react";

export const GifGrid = ({ category }) => {
  const [count, setCount] = React.useState(0);

  React.useEffect(() => {
    getGifs();
  });

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=Uribe&limit=10&api_key=Enhu5gM26OHBLH0hYgf1tKNVcT2JHheo";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(({ id, title, images }) => {
      return {
        id: id,
        title: title,
        url: images?.downsized_medium.url,
      };
    });

    console.log(`gifs`, gifs);
  };
  return (
    <div>
      <h3>{category}</h3>

      <h3>{count}</h3>
      <button
        onClick={() => {
          setCount(count + 1)}}
      ></button>
    </div>
  );
};
