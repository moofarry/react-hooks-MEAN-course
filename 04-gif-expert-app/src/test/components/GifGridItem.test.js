import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("test in GifGridItem", () => {
  const title = "one title";
  const url = "https://localhost/something.jpeg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should be show the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a paragraph with the title  ", () => {
    const p = wrapper.find('p');
    expect(p.text().trim()) .toBe(title);
  });

  test('should have img and url with props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  })

  test('should have animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('card')).toBe(true);
  })
  
  
});
