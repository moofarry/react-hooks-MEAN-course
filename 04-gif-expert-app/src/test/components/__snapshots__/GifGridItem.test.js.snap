// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGridItem should be show the component 1`] = `
<div
  className="card "
>
  <img
    alt="one title"
    src="https://localhost/something.jpeg"
  />
  <p>
     
    one title
  </p>
</div>
`;
